app/controllers/application_controller.rb

```diff
@@ -141,6 +141,17 @@ def user_id_mismatch
     })
   end
 
+  def not_authorized_run?(run)
+    begin
+      authorize! :access, @run
+    rescue
+      user_id_mismatch()
+      render 'runs/unauthorized_run', status: :unauthorized
+      return true
+    end
+    return false
```

Comment:
It would be better to throw an exception here instead of returning. This way callers of it don't need to use the form `return if not_authorized_run?`. However I'm not sure how that will interact with the render call. So it might be necessary to add an exception handler which does the actual rendering of 'runs/unauthorized_run'. 

This is done using a approach like https://github.com/concord-consortium/lara/blob/01046ef43b963c136c86b4a3bcd9eae8f972a769/app/controllers/application_controller.rb#L12-L13  So we'd have a custom exception and a rescue_from which catches that exception.

Perhaps it might even be best if the existing CanCan:AccessDenied handler just checks for a action of 'access' and a subject of 'Run' (or whatever strings or symbols should be used there).  And then does what is in not_authorized_run? is doing.  Then callers just need to do `authorize! :access, @run` .